controllers/neutronapi_controller.go

```diff
@@ -448,8 +449,21 @@ func (r *NeutronAPIReconciler) reconcileInit(
 			},
 		)
 		if err != nil {
+			instance.Status.Conditions.Set(condition.FalseCondition(
+				condition.TLSInputReadyCondition,
+				condition.ErrorReason,
+				condition.SeverityWarning,
+				condition.TLSInputErrorMessage,
+				err.Error()))
 			return ctrlResult, err
 		} else if (ctrlResult != ctrl.Result{}) {
+			// Marking the condition as Unknown because we are not returining
+			// an err, but comparing the ctrlResult: this represents an in
+			// progress operation rather than something that failed
+			instance.Status.Conditions.MarkUnknown(
+				condition.TLSInputReadyCondition,
+				condition.InitReason,
+				condition.InputReadyInitMessage)
```

Comment:
I think it would be better to use these [1] [2] here.

[1] https://github.com/openstack-k8s-operators/lib-common/blob/56015b1ae3f62721195798f7564f0abcba937547/modules/common/condition/conditions.go#L86-L87
[2] https://github.com/openstack-k8s-operators/lib-common/blob/56015b1ae3f62721195798f7564f0abcba937547/modules/common/condition/conditions.go#L137-L138

Reply:
Makes sense, updating that!